### DISCUSSION TITLE
How to put validation message around 3rd party components?

### DISCUSSION THREAD

**Discussion by sturlath** [Megabit/Blazorise#3272] [Q&A] [answered]
I'm using (for now) this 3rd party upload component and I need to validate that I have uploaded a file.

And I have it working like this

![image](https://user-images.githubusercontent.com/2386572/147754080-64fc7f67-8790-44aa-8770-093b288612ef.png)

My question is, **Is this (here below) the only way to do this kind of validation?** 

That is to use e.g. `<TextEdit>` to have access to `<Feedback>` and then adding [Visibility.Invisible](https://github.com/Megabit/Blazorise/blob/67da3e1fb3256f83f25eb378026eda10768322b5/Source/Blazorise/Enums/Visibility.cs) and flexing the message up!

Or am I missing another way? At least somebody can hopefully benefit from this question.

```xml
<Validations @ref="validations" Mode="ValidationMode.Manual">
    <Validation Validator="@ValidateRecordingUpload">
        <Field>
            <FieldBody>
                <FileUpload3rdParty >
                </FileUpload3rdParty >
            </FieldBody>
            <Div Flex="Flex.JustifyContent.Start.AlignItems.Baseline" Position="Position.Relative"  >
                <Div Position="Position.Absolute.Top.Is0.Start.Translate.MiddleY">
                    <TextEdit Visibility="Visibility.Invisible">
                        <Feedback>
                            <ValidationError />
                        </Feedback>
                    </TextEdit>
                </Div>
            </Div>
        </Field>
    </Validation>
</Validations>
```

Btw I thought that `Visibility.None` would hide the element so that it wouldn´t take up space but if I use that this here is the result (yes with the flex..)

![image](https://user-images.githubusercontent.com/2386572/147754914-ca1a9bc9-2149-499a-a178-df3bdafde4f2.png)

P.s
When I was searching for `Visibility` in the docs I didn´t find it and see that there are [loads of enums missing](https://github.com/Megabit/Blazorise/tree/67da3e1fb3256f83f25eb378026eda10768322b5/Source/Blazorise/Enums). I would help with adding them to the documentation if I wasn't super pressed for time atm.

**Comment by David-Moreira** [comment, marked as answer]
This comes up a few times. haha

Visibility is exactly as the name says, the html component is there, rendered in the html, but invisible!! :D
You should use `display` to control whether you want it to render or not. so try setting `display:none`, it should actually make it so, it is not rendered and therefore not "take up the space"

**Reply by David-Moreira** [reply]
Adding similar posted issue: https://github.com/Megabit/Blazorise/issues/3217

**Reply by David-Moreira** [reply]
And I guess we can add the enum if it's missing and a `Note` for this html behaviour as not everyone might know it by heart.

**Reply by sturlath** [reply]
Ah... I swear that I looked first for Display in the docs but there I saw Visibility and none... and got stuck there :-) 
But adding some mentioning to Display in Visibility docs (and visa versa) could clear up the confusion I think.

But great job and again sorry for the spam! At least there is another example of using Flex hehe
